tests/test_cli_run.py

```diff
@@ -32,10 +32,52 @@ def test_mutually_exclusive_params(tmp_path):
     cmd = ['run', str(tmp_path / 'main.wdl'), '--docker', '--singularity']
     with pytest.raises(ValueError):
         cli_main(cmd)
+
+    cmd = [
+        'run',
+        str(tmp_path / 'main.wdl'),
+        '--docker',
+        'ubuntu:latest',
+        '--singularity',
+    ]
+    with pytest.raises(ValueError):
+        cli_main(cmd)
+
+    cmd = [
+        'run',
+        str(tmp_path / 'main.wdl'),
+        '--docker',
+        '--singularity',
+        'docker://ubuntu:latest',
+    ]
+    with pytest.raises(ValueError):
+        cli_main(cmd)
+
+    cmd = [
+        'run',
+        str(tmp_path / 'main.wdl'),
+        '--docker',
+        'ubuntu:latest',
+        '--singularity',
+        'docker://ubuntu:latest',
+    ]
+    with pytest.raises(ValueError):
+        cli_main(cmd)
+
     cmd = ['run', str(tmp_path / 'main.wdl'), '--docker', '--soft-glob-output']
     with pytest.raises(ValueError):
         cli_main(cmd)
 
+    cmd = [
+        'run',
+        str(tmp_path / 'main.wdl'),
+        '--docker',
+        'ubuntu:latest',
+        '--soft-glob-output',
+    ]
+    with pytest.raises(ValueError):
+        cli_main(cmd)
+
```

Comment:
This seems repetitive. You can use `pytest.mark.parametrize` to parametrize the test over different `cmd` lists, see https://docs.pytest.org/en/stable/parametrize.html

Reply:
Thanks, replace repetitive lines with `pytest.mark.parametrize`.